database/migrations/sql/20190220_19_add_temp_region_sector.sql

```diff
@@ -0,0 +1,6 @@
+/*
+ * Migration template
+ */
+
+INSERT INTO community_business_sector (sector_name) VALUES ('TEMPORARY DATA');
```

Comment:
Why is this necessary?

Reply:
`sector_id` is a required foreign key for `community_business`. I wanted to attach it to its own row to avoid complications with aggregated statistics in the future

Happy for suggestions on a better name.

Reply:
It's better to represent things directly than indirectly. That means things can be declared rather than inferred.

If temp/dummy organisations are a long term feature, then we want them to be segregated by default from real organisations. Ideally that would mean the temp/dummy nature of an organisation should be a first-class property of the organisation model.

At the moment we're inferring that an org is temp by the name/email. This is a pragmatic step to avoid changing the data model, but I don't think that should be a long-term thing.

We should go through the CB model methods and make sure "normal" methods cannot return "temp" orgs, and "temp" methods cannot return "normal" orgs (and similarly for cb admin methods).

Reply:
Is this in addition to having a temp value for sectors/regions? I do not think we need to collect this info for creating a temp account and I do not want to just throw them under an arbitrary value.

Reply:
> Is this in addition to having a temp value for sectors/regions?

It's instead of.

> I do not want to just throw them under an arbitrary value.

Why not? They only reason regions/sectors are needed is in order to be able to calculate statistics across and within these categories. If we can already segregate out the "fake" orgs, then the quality of the statistics is safe.

Reply:
> Why not?

I do not feel that we should be storing data inaccurately in the db if it can be avoided. What do we loose by having temp values in region/sector?

Reply:
Every aspect of these temp organisations is inaccurate

Reply:
None of the data added is fake though.

Reply:
Still keen to know:

> What do we loose by having temp values in region/sector?

Reply:
> None of the data added is fake though.

What do you mean by "fake"? It's all intended for eventual deletion, none of the data associated to those temp accounts matters at all.

> What do we loose by having temp values in region/sector?

We _gain_ cruft, indirection, and workarounds.

I don't want to hold this up too much. I don't like this idea, but if you really want to leave them in, do.

But I would like this to be done:

> We should go through the CB model methods and make sure "normal" methods cannot return "temp" orgs, and "temp" methods cannot return "normal" orgs (and similarly for cb admin methods).

It makes sense for the definition of "temporary" account to be abstracted out somewhere so it can be easily changed in future.

I will create an issue about adding in a field to identify temp accounts/orgs directly (and relatedly, discussing what will happen to the `subscription` and `funding_body` tables), and then updating the temp CB model methods.

Reply:
> What do you mean by "fake"?

Here's an example of all org/cb data we hold for a temp cb created as part of this PR. Any data that is not relevant for a temp account is set as `null`. I cannot set `region_id` & `sector_id` as `null` because of db constraints, so they have their own related row `TEMPORARY DATA` inserted into each table. 

```js
organisation_id: 14,
    organisation_name: 'TEMPORARY ACCOUNT: bla',
    _360_giving_id: null,
    created_at: 2019-02-22T17:24:20.710Z,
    modified_at: null,
    deleted_at: null,
    community_business_region_id: 10,
    community_business_sector_id: 15,
    address_1: null,
    address_2: null,
    town_city: null,
    post_code: null,
    coordinates: null,
    logo_url: null,
    turnover_band: null }
```

> But I would like this to be done:

In this PR?

Reply:
> In this PR?

I was thinking yes, but actually maybe it's not that important yet. Can you think of any places where it might cause an issue for temporary and regular orgs to be treated exactly the same?

* Login - fine to treat them the same
* `GET /community-businesses/{id?}` is only for use by `TWINE_ADMINS` so far, so that's not a big deal
* `<ANY> community-businesses/me/{param?}` - fine already

Anything else?

Reply:
OK, then I'll mark this resolved once those issues are created.

Reply:
Created #361 #362 #363